token-tracker/src/lib/uniswap.js

```diff
@@ -0,0 +1,38 @@
+import axios from "axios"
+
+const UNISWAP_API_URL =
+  "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2"
+
+export const getPairData = async (pairId) => {
+  const response = await axios.post(UNISWAP_API_URL, {
```

Comment:
@pdyraga please note, that we always fetch current state of `pairData`. If the "target_block" is in the past when we run this code, then we still calculate KEEP amount in LP tokens based on the current state of the pools. We can easily change that by passing a "target_block" to this query.  The question is whether we want calculation based on the "target_block" or the current state?

Reply:
I think we have to check it at the target block.

Reply:
`Target block` indeed then.